dll/shellext/netshell/lanstatusui.cpp

```diff
@@ -354,6 +356,45 @@ InsertItemToListView(
     return -1;
 }
 
+static BOOL tmToStr(
+    IN struct tm *pTM,
+    OUT LPWSTR szBuffer,
+    IN UINT nBufferSize)
+{
+    SYSTEMTIME st;
+    LPWSTR szBufferDate;
+    LPWSTR szBufferTime;
+    UINT nCharDate, nCharTime;
+    BOOL bResult;
+
+    bResult = FALSE;
+    st.wYear = pTM->tm_year + 1900;
+    st.wMonth = pTM->tm_mon + 1;
+    st.wDay = pTM->tm_mday;
+    st.wHour = pTM->tm_hour;
+    st.wMinute = pTM->tm_min;
+    st.wSecond = pTM->tm_sec;            
+    
+    //Check required size before cpy/cat
+    nCharDate = GetDateFormatW(LOCALE_USER_DEFAULT, 0, &st, nullptr, NULL, 0);
+    nCharTime = GetTimeFormatW(LOCALE_USER_DEFAULT, 0, &st, nullptr, NULL, 0);
+    
+    szBufferDate = (LPWSTR)HeapAlloc(GetProcessHeap(), HEAP_ZERO_MEMORY, (nCharDate+1) * sizeof(WCHAR));
+    szBufferTime = (LPWSTR)HeapAlloc(GetProcessHeap(), HEAP_ZERO_MEMORY, (nCharTime+1) * sizeof(WCHAR));	
+    if ((szBufferDate == NULL) || (szBufferTime == NULL)) return FALSE;
```

Comment:
```suggestion
    if (!szBufferDate)
        return FALSE;
    szBufferTime = (LPWSTR)HeapAlloc(GetProcessHeap(), HEAP_ZERO_MEMORY, (nCharTime+1) * sizeof(WCHAR));
    if (!szBufferTime)
    {
        HeapFree(GetProcessHeap(), 0, szBufferDate);
        return FALSE;
    }
```

Reply:
Also since I see that netshell includes atlstr.h (from the precompiled header), I would suggest using the CAtlStrings instead. @learn-more can you provide some hints?

Reply:
That sounds like a very good improvement.
See https://git.reactos.org/?p=reactos.git;a=blob;f=base/applications/mspaint/registry.cpp;hb=4247e3f838ef32f209e51a278114bcdb78eea59e#l31 for an example.

Reply:
Good point on the memory leak. Fixed.
What's the benefit of locally using CString here ?

Reply:
All comments take into account except the on the use of CString. Please let me know if you consider this as a "must" for this PR.

Reply:
The usage of CString is usually interesting as it'll manage for you the memory allocation and deallocation when execution goes out of scope

Reply:
Now fully relying on CString.